person.rb

```diff
@@ -0,0 +1,40 @@
+require_relative 'decorators'
+class Person < Nameable
+  attr_reader :id
+  attr_accessor :name, :age, :parent_permission
+
+  def initialize(age, name = 'Unknown', parent_permission: true)
+    super()
+    @id = Random.rand(1..1000)
+    @name = name
+    @age = age
+    @parent_permission = parent_permission
+    @rentals = []
+  end
+
+  def of_age?
+    @age >= 18
+  end
+
+  def can_use_services?
+    of_age? || @parent_permission
+  end
+
+  def correct_name
+    @name
+  end
+
+  def add_rentals(rental)
+    @rentals.push(rental)
+    rental.person = self
+  end
+
+  private :of_age?
+end
+
+# person = Person.new(22, 'maximilianus')
+# person.correct_name
+# capitalized_person = CapitalizeDecorator.new(person)
+# puts capitalized_person.correct_name
+# capitalized_trimmed_person = TrimmerDecorator.new(capitalizedPerson)
+# puts capitalized_trimmed_person.correct_name
```

Comment:
- [ ] **Optional**, Please remove all commented code from your codebase to make it clean and easy to maintain.